src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs

```diff
@@ -14,57 +13,41 @@
 using Microsoft.CodeAnalysis.Options;
 using Microsoft.CodeAnalysis.Simplification;
 using Microsoft.DotNet.ApiSymbolExtensions;
-using Microsoft.DotNet.ApiSymbolExtensions.Filtering;
-using Microsoft.DotNet.ApiSymbolExtensions.Logging;
 using Microsoft.DotNet.GenAPI.SyntaxRewriter;
 
 namespace Microsoft.DotNet.GenAPI
 {
     /// <summary>
     /// Processes assembly symbols to build corresponding structures in C# language.
     /// </summary>
-    public sealed class CSharpFileBuilder : IAssemblySymbolWriter, IDisposable
+    public sealed class CSharpFileBuilder : IAssemblySymbolWriter
     {
-        private readonly ILog _logger;
+        private readonly GenApiAppConfiguration _c;
         private readonly TextWriter _textWriter;
-        private readonly ISymbolFilter _symbolFilter;
-        private readonly ISymbolFilter _attributeDataSymbolFilter;
-        private readonly string? _exceptionMessage;
-        private readonly bool _includeAssemblyAttributes;
         private readonly AdhocWorkspace _adhocWorkspace;
         private readonly SyntaxGenerator _syntaxGenerator;
-        private readonly IEnumerable<MetadataReference> _metadataReferences;
-
-        public CSharpFileBuilder(ILog logger,
-            ISymbolFilter symbolFilter,
-            ISymbolFilter attributeDataSymbolFilter,
-            TextWriter textWriter,
-            string? exceptionMessage,
-            bool includeAssemblyAttributes,
-            IEnumerable<MetadataReference> metadataReferences)
+
+        public CSharpFileBuilder(GenApiAppConfiguration configuration, TextWriter textWriter)
```

Comment:
Before we were passing-in explicitly what the class needed. Why do we need to change that?

Reply:
Hmmm well, I see your point: on one hand, I don't intend to change the behavior of this class (I will create a new one), so the arguments don't need to be changed. I can still pass only what the class needs. On the other hand, passing the whole configuration object is handy.

I can revert this to only take the arguments it needs. I don't think it will affect the new diff code.

Reply:
Yes. Done.